src/core/prototypes/state.ts

```diff
@@ -153,12 +153,17 @@ export class ChartStateManager {
       for (let index = 0; index < origin.constraints.length; index++) {
         const originConstringts = origin.constraints[index];
         const current = chart.constraints[index];
-        expect_deep_approximately_equals(
-          originConstringts,
-          current,
-          defaultDifferenceApproximation,
-          true
-        );
+        try {
+          expect_deep_approximately_equals(
+            originConstringts,
+            current,
+            defaultDifferenceApproximation,
+            true
+          );
+        } catch (ex) {
+          console.log(ex);
```

Comment:
we probably don't need this console log, it just means the objects are different and we have changes to save, right?

Reply:
Yes. It is